File: Question/49_Group_Anagrams.cpp
class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
        vector<vector<string>> ans;
        unordered_map<string, vector<string>> mem;

        for(auto& str : strs) {
            string key = str;
            sort(key.begin(), key.end());
            mem[key].push_back(str);
        }

        for(auto& i : mem) {
            ans.push_back(i.second);
        }
        return ans;
    }
};

File: Question_Java/49_Group_Anagrams.java
class Solution {
    public List<List<String>> groupAnagrams(String[] strs) {
        HashMap<String, List<String>> mem = new HashMap<>();
        for (String str : strs) {
            char[] chs = str.toCharArray();
            Arrays.sort(chs);
            String temp = new String(chs);
            List<String> l = mem.getOrDefault(temp, new ArrayList<String>());
            l.add(str);
            mem.put(temp, l);
        }
        return new ArrayList<List<String>>(mem.values());
    }
}